bluepysnap/bbp.py

```diff
@@ -17,7 +17,217 @@
 
 """BBP cell / synapse attribute namespace."""
 
-from bluepysnap.sonata_constants import DYNAMICS_PREFIX, Edge, Node
+from bluepysnap.sonata_constants import DYNAMICS_PREFIX, ConstContainer, Edge, Node
+
+NODE_TYPES = ['biophysical', 'virtual', 'astrocyte', 'single_compartment', 'point_neuron']
+EDGE_TYPES = ['chemical', 'electrical', 'synapse_astrocyte', 'endfoot']
+
+
+class Astrocyte(ConstContainer):
```

Comment:
They're found in the [Circuit Documentation](https://bbpteam.epfl.ch/documentation/projects/Circuit%20Documentation/latest/sonata_tech.html), and these basically represent the ones listed as mandatory there. These are actually currently not used but my idea was to check the files for these fields in `circuit_validation` and then throw an error if any of them is missing. 

I would also expect them to inherit the `Node` (or `Edge` in case of edges) but these don't all have the same attributes defined in `sonata_constants`, so I am not sure what would be the best practice here.

Reply:
I'm hesitating here. I see so much code duplication among those enums. @mgeplf what are your thoughts here?

Reply:
If this code is not actually used (yet), we could move it into a different PR to be completed with the validation code.
Maybe it will be clearer how to organize it, and maybe we'll want to validate both mandatory and optional fields.

Reply:
Actually, I think this is used in the ArchNGV PR by @asanin-epfl  and @eleftherioszisis

Reply:
ArchNGV only redirects to those classes. It imports those instances so users of ArchNGV can import them like, it does not use them in its code:
```python
from archngv.core.constants import Astrocyte, Cell, Synapse.
```
https://bbpgitlab.epfl.ch/molsys/ArchNGV/-/merge_requests/21/diffs#106c97ac08ff01bdaf1c3533d4abd91c5b1ac243_21_6

Reply:
I see. If you guys rather not have them defined in this PR, I'm fine not having them. Regarding the model type validation, should I just validate it by comparing to the value in the `*.h5` files then (instead of comparing to what is available)?

Reply:
Lets not add them yet then. I didn't understand clearly the proposition with `*.h5` files. Can you please elaborate? May be a small example?

Reply:
Ah, yeah. My idea was to compare it to the `model_type` in the `h5` file, but I just realized that field is not present in the edge files. So I guess, at least for now, we skip the validation, too. Any ideas?

Reply:
> Any ideas?

No ideas. I support your proposition that we can skip this validation.

Reply:
As there's no objections, I remove the unused types and don't validate the model type